apps/src/templates/rubrics/LearningGoals.jsx

```diff
@@ -0,0 +1,392 @@
+        autosave();
+      }, saveAfter);
+    }
+  };
+
+  const getAiUnderstanding = learningGoalId => {
+    if (!!aiEvaluations) {
+      const aiInfo = aiEvaluations.find(
+        item => item.learning_goal_id === learningGoalId
+      );
+      return aiInfo?.understanding;
+    } else {
+      return null;
+    }
+  };
+
+  const getAiConfidence = learningGoalId => {
+    if (!!aiEvaluations) {
+      const aiInfo = aiEvaluations.find(
+        item => item.learning_goal_id === learningGoalId
+      );
+      return aiInfo?.ai_confidence;
+    } else {
+      return null;
+    }
+  };
+
+  const getAiInfo = learningGoalId => {
+    if (!!aiEvaluations) {
+      const aiInfo = aiEvaluations.find(
+        item => item.learning_goal_id === learningGoalId
+      );
+      return aiInfo;
+    } else {
+      return null;
+    }
+  };
+
+  const aiUnderstanding = getAiUnderstanding(
+    learningGoals[currentLearningGoal].id
+  );
+  const aiConfidence = getAiConfidence(learningGoals[currentLearningGoal].id);
+  const aiEvalInfo = getAiInfo(learningGoals[currentLearningGoal].id);
+
+  const autosave = () => {
+    setAutosaved(false);
+    setIsAutosaving(true);
+    setErrorAutosaving(false);
+    const bodyData = JSON.stringify({
+      studentId: studentLevelInfo.user_id,
+      learningGoalId: learningGoals[currentLearningGoal].id,
+      feedback: teacherFeedback.current,
+      understanding: understandingLevel.current,
+    });
+    HttpClient.put(
+      `${base_teacher_evaluation_endpoint}/${learningGoalEval.id}`,
+      bodyData,
+      true,
+      {
+        'Content-Type': 'application/json',
+      }
+    )
+      .then(() => {
+        setIsAutosaving(false);
+        setAutosaved(true);
+        if (!feedbackAdded) {
+          setFeedbackAdded(true);
+        }
+      })
+      .catch(error => {
+        console.log(error);
+        setIsAutosaving(false);
+        setErrorAutosaving(true);
+      });
+    clearTimeout(autosaveTimer.current);
+  };
+
+  useEffect(() => {
+    if (studentLevelInfo && learningGoals[currentLearningGoal].id) {
+      const body = JSON.stringify({
+        userId: studentLevelInfo.user_id,
+        learningGoalId: learningGoals[currentLearningGoal].id,
+      });
+      HttpClient.post(
+        `${base_teacher_evaluation_endpoint}/get_or_create_evaluation`,
+        body,
+        true,
+        {
+          'Content-Type': 'application/json',
+        }
+      )
+        .then(response => response.json())
+        .then(json => {
+          setLearningGoalEval(json);
+          if (json.feedback) {
+            teacherFeedback.current = json.feedback;
+            setDisplayFeedback(teacherFeedback.current);
+          }
+          if (json.understanding >= 0 && json.understanding !== null) {
+            setDisplayUnderstanding(json.understanding);
+            understandingLevel.current = json.understanding;
+          }
+        })
+        .catch(error => console.log(error));
```

Comment:
`console.error` here too?

Reply:
in this case, I'm not sure there is any need to catch at all if all you care about is letting the error show up to the user and our monitoring.